File: 2.Kevoree_Basics/1.HelloWorld/java/src/main/java/org/kevoree/example/HelloWorld.java
package org.kevoree.example;

import org.kevoree.annotation.ComponentType;
import org.kevoree.annotation.Start;
import org.kevoree.annotation.Stop;
import org.kevoree.log.Log;

@ComponentType
public class HelloWorld {

    @Start
    public void start() {
        Log.info("Hello, world!");
    }

    @Stop
    public void stop() {
        Log.info("Bye, world!");
    }
}

File: 2.Kevoree_Basics/6.Contrib/Tellu/YeomanTest/lib/HelloWorld.js
var AbstractComponent = require('kevoree-entities').AbstractComponent;

/**
 * Kevoree component
 * @type {HelloWorld}
 */
var HelloWorld = AbstractComponent.extend({
    toString: 'HelloWorld',
            
    /* This is an example of dictionary attribute that you can set for your entity */
    dic_yourAttrName: {
        optional: false,
        defaultValue: 'aDefaultValue',
    },

    /**
     * this method will be called by the Kevoree platform when your component has to start
     * @param {Function} done
     */
    start: function (done) {
        this.log.debug(this.toString(), 'START');
        done();
    },

    /**
     * this method will be called by the Kevoree platform when your component has to stop
     * @param {Function} done
     */
    stop: function (done) {
        this.log.debug(this.toString(), 'STOP');
        done();
    }
    
});

module.exports = HelloWorld;
